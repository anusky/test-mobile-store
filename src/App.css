@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .breadcrumb-item {
    @apply text-lg text-white;
  }
  a.breadcrumb-item {
    @apply font-medium;
  }
  .breadcrumb-item--last {
    @apply text-slate-200;
  }
}
