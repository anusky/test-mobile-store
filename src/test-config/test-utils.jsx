import "./__mocks__/intersectionObserverMock";

import { render } from "@testing-library/react";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import PropTypes from "prop-types";

import productInfo from "./__fixtures__/product.json";
import productList from "./__fixtures__/productList.json";
import cart from "./__fixtures__/cart.json";
import { ProductContextProvider } from "src/context/productContext";
import { PAGE_STRUCTURE } from "src/utils/constants";

const customRender = (ui, options = {}) => {
  const { routerInitialIndex = 0 } = options;

  const AllTheProviders = ({ children }) => {
    const routes = [
      {
        path: "/",
        element: children,
        id: PAGE_STRUCTURE.home.clientPageId,
        loader: () => ({
          productList,
          cart,
        }),
        handle: {
          crumb: () => ({ url: "/", content: "home route" }),
        },
        children: [
          {
            path: "/:productId",
            element: children,
            id: PAGE_STRUCTURE.product.clientPageId,
            loader: () => productInfo,
            handle: {
              crumb: () => ({ url: "/", content: "product route" }),
            },
          },
        ],
      },
    ];

    const router = createMemoryRouter(routes, {
      initialEntries: ["/", "/123"],
      initialIndex: routerInitialIndex,
    });
    return (
      <RouterProvider router={router}>
        <ProductContextProvider
          value={{
            list: productList,
            cart: cart,
          }}
        >
          {children}
        </ProductContextProvider>
      </RouterProvider>
    );
  };

  AllTheProviders.propTypes = {
    children: PropTypes.node.isRequired,
  };

  return render(ui, { wrapper: AllTheProviders, ...options });
};

export * from "@testing-library/react";
export { customRender as render };
