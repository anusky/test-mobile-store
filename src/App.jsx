import { Outlet, useRouteLoaderData } from "react-router-dom";

import "./App.css";
import { ProductContextProvider } from "./context/productContext";
import { PAGE_STRUCTURE } from "./utils/constants";

import { getAlProducts } from "./utils/productManagement";

export const appLoader = async () => {
  return await getAlProducts();
};

function App() {
  const productInfo = useRouteLoaderData(PAGE_STRUCTURE.home.clientPageId);

  return (
    <ProductContextProvider value={{ list: productInfo }}>
      <div className="App">
        <header className="App-header">
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>

        <Outlet />
      </div>
    </ProductContextProvider>
  );
}

export default App;
