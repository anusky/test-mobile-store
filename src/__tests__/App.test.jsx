import { act } from "react-dom/test-utils";
import { render, screen } from "src/test-config/test-utils";
import App from "../App";

describe("App coverage", () => {
  it("renders learn react link", async () => {
    render(<App />);

    expect(await screen.findByText(/learn react/i)).toBeInTheDocument();
  });

  it("matches snapshot", async () => {
    let container;
    await act(() => {
      container = render(<App />);
    });
    expect(container.asFragment()).toMatchSnapshot();
  });
});
