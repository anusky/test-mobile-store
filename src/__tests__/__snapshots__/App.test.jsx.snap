// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App coverage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</DocumentFragment>
`;
